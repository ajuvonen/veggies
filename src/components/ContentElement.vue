<script lang="ts" setup>
withDefaults(
  defineProps<{
    title: string;
    ariaHidden?: boolean;
  }>(),
  {
    ariaHidden: false,
  },
);
</script>
<template>
  <div class="flex-container flex-col">
    <h1 class="label-like" :aria-hidden="ariaHidden">{{ title }}</h1>
    <slot />
  </div>
</template>
