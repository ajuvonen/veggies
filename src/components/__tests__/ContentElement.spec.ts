import {describe, it, expect} from 'vitest';
import {mount} from '@vue/test-utils';
import ContentElement from '@/components/ContentElement.vue';

describe('ContentElement', () => {
  it('renders', () => {
    const wrapper = mount(ContentElement, {
      props: {
        title: 'Test',
      },
      slots: {default: '<div>Content</div>'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders title as label', () => {
    const wrapper = mount(ContentElement, {
      props: {
        title: 'Test',
        labelTag: 'label',
      },
    });

    expect(wrapper.find('label').text()).toBe('Test');
  });

  it('passes attributes to label', () => {
    const wrapper = mount(ContentElement, {
      props: {
        title: 'Test',
        labelAttrs: {'aria-hidden': true},
      },
    });

    expect(wrapper.find('h2').attributes('aria-hidden')).toBe('true');
  });

  it('passes attributes to container', () => {
    const wrapper = mount(ContentElement, {
      props: {
        title: 'Test',
        containerAttrs: {
          style: 'height: 100%;',
        },
      },
    });

    expect(wrapper.find('div').attributes('style')).toEqual('height: 100%;');
  });

  it('merges classes', () => {
    const wrapper = mount(ContentElement, {
      props: {
        title: 'Test',
        labelAttrs: {
          class: 'truncate',
        },
      },
    });

    expect(wrapper.find('h2').classes()).toEqual(['label-like', 'truncate']);
  });

  it('merges classes with attribute', () => {
    const wrapper = mount(ContentElement, {
      props: {
        title: 'Test',
      },
      attrs: {
        class: 'min-h-0',
      },
    });

    expect(wrapper.find('div').classes()).toEqual(['flex-container', 'flex-col', 'min-h-0']);
  });
});
