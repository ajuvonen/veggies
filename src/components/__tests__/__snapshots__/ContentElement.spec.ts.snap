// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentElement > renders 1`] = `
"<div class="flex-container flex-col">
  <h2 class="label-like">Test</h2>
  <div>Content</div>
</div>"
`;
